src/lib/emitters.rs

```diff
@@ -85,6 +94,151 @@ pub fn write_json_to_stream<T: RenderJson, U: RenderJson>(mut stream: Box<Write>
         .map_err(|_| EmitterError::CouldNotWriteToFile)
 }
 
+// Map action types to terminal colours.
+fn build_colour_map() -> HashMap<ActionType, term::color::Color> {
+    let mut map: HashMap<ActionType, term::color::Color> = HashMap::new();
+    map.insert(ActionType::DELETE, term::color::BRIGHT_RED);
+    map.insert(ActionType::INSERT, term::color::BRIGHT_GREEN);
+    map.insert(ActionType::MOVE, term::color::BRIGHT_BLUE);
+    map.insert(ActionType::UPDATE, term::color::BRIGHT_YELLOW);
+    map
+}
+
+// Read file and return its contents or `ParseError`.
+fn read_file(path: &str) -> Result<String, EmitterError> {
+    let mut f = match File::open(path) {
+        Ok(r) => r,
+        Err(_) => return Err(EmitterError::CouldNotOpenFile),
+    };
+    let mut s = String::new();
+    f.read_to_string(&mut s).unwrap();
+    Ok(s)
+}
+
+/// Write diff to STDOUT, highlighting actions in the edit script.
+///
+/// This function prints the whole file, rather than just the relevant context.
+/// Patch format is not supported, and hunk headers (e.g. `@@ -1 +1 @@`) are
+/// not printed.
+///
+/// ## Panics
+/// If the terminal foreground cannot be changed or reset by the
+/// [term](https://crates.io/crates/term) crate, this function will panic.
+/// However, `term` is cross-platform so this failure case should be rare.
+pub fn write_colour_diff_to_stdout<T: Clone + Display + Eq>(store: &MappingStore<T>,
```

Comment:
Can we maybe find a snappier name for the function? It's a real mouthful and perhaps a bit over-specific.